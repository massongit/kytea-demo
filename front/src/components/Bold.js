import React from "react"
import PropTypes from "prop-types"

/**
 * 文章を強調するためのComponent
 */
class Bold extends React.Component {
    static propTypes = {
        children: PropTypes.node.isRequired
    }

    render() {
        return (
            <strong>
                <u>
                    {this.props.children}
                </u>
            </strong>
        )
    }
}

export default Bold
