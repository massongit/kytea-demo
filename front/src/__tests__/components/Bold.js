import React from "react"
import Bold from "../../components/Bold"
import {shallow} from "enzyme"

const text = "test"

const div = (
    <div>
        {text}
    </div>
)

const lineComponentDiv = shallow(
    <Bold>
        {div}
    </Bold>
)

describe("components/Bold", () => {
    it("子要素として与えられたElementが正しく挿入されている", () => {
        expect(lineComponentDiv).toMatchSnapshot()
        expect(lineComponentDiv.children().contains(div)).toBeTruthy()
    })

    it("子要素として与えられた文字列が正しく挿入されている", () => {
        const lineComponentText = shallow(
            <Bold>
                {text}
            </Bold>
        )
        expect(lineComponentText).toMatchSnapshot()
        expect(lineComponentText.children().contains(text)).toBeTruthy()
    })
})
