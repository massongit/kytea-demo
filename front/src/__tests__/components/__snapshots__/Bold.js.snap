// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Bold 子要素として与えられたElementが正しく挿入されている 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bold>
    <div>
      test
    </div>
  </Bold>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <u>
        <div>
          test
        </div>
      </u>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          test
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
      "type": "u",
    },
    "type": "strong",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <u>
          <div>
            test
          </div>
        </u>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            test
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "div",
        },
        "type": "u",
      },
      "type": "strong",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`components/Bold 子要素として与えられた文字列が正しく挿入されている 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bold>
    test
  </Bold>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <u>
        test
      </u>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
      },
      "ref": null,
      "rendered": "test",
      "type": "u",
    },
    "type": "strong",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <u>
          test
        </u>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "u",
      },
      "type": "strong",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
